http4s/src/main/scala/caliban/Http4sAdapter.scala

```diff
@@ -95,6 +95,19 @@ object Http4sAdapter {
     }
   }
 
+  def executeRequest[R0, R, Q, M, S, E](interpreter: GraphQL[R, Q, M, S, E], provideEnv: R0 => R): HttpApp[RIO[R0, *]] =
+    Kleisli { req =>
+      object dsl extends Http4sDsl[RIO[R0, *]]
+      import dsl._
+      for {
+        query <- req.attemptAs[GraphQLRequest].value.absolve
```

Comment:
Can you extract this block into a function so that it's not repeated here and in `makeRestService`?
```scala
query <- req.attemptAs[GraphQLRequest].value.absolve
result <- execute(interpreter, query)
                   .foldCause(cause => GraphQLResponse(NullValue, cause.defects).asJson, _.asJson)
```

Reply:
👍

Reply:
I looks like good idea but in executeRequest I work with Request/Response with environment R0 and executor is in environment R, so it would mean doing two `mapK` which I do not like much and I find it now more simple. Or I can to common method `provideSome` and send `indentity` from `makeRestService` which also not nice. What do you think?

Reply:
Oh I see what you mean. I'd prefer to avoid an unnecessary `provideSome` for general case. How about just moving `execute(interpreter, query).foldCause(cause => GraphQLResponse(NullValue, cause.defects).asJson, _.asJson)` to a function?